### PR TITLE
add code folders to analytics projects 

### PR DESCRIPTION
**Pull request by smithgp**
### What does this PR do?
In `force:project:create -t analytics`:
* Include `aura/`, `classes/`, and `lwc/` folders.
* Include the aura and lwc eslint configs.
* Adds the analytics vscode extension to the extension recommendations.

Also, updates the node version in `.nvmrc` to match the volta config entry.

### What issues does this PR fix or reference?
@W-9191555@

### CODE REVIEW

**Review by reidaelliott — APPROVED** [review]

File: packages/templates/src/generators/projectGenerator.ts [hunk 1]
```diff
@@ -124,30 +123,14 @@ export default class ProjectGenerator extends SfdxGenerator<ProjectOptions> {
       this.fs.copyTpl(
         this.templatePath('lwc.eslintrc.json'),
         this.destinationPath(
-          path.join(
```
Comment (smithgp) [inline packages/templates/src/generators/projectGenerator.ts:127]: This is just removing some duplicate code used in a number of places (`folderlayout` was already defined to point to the `default` folder path).

File: packages/templates/src/generators/projectGenerator.ts [hunk 2]
```diff
@@ -175,7 +158,6 @@ export default class ProjectGenerator extends SfdxGenerator<ProjectOptions> {
       }
     }
 
-    // tslint:disable-next-line:no-unused-expression
```
Comment (smithgp) [inline packages/templates/src/generators/projectGenerator.ts:178]: These tslint disables don't appear to be needed any longer.

File: .nvmrc [hunk 3]
```diff
@@ -1 +1 @@
-v10.15.3
\ No newline at end of file
+v12.4.0
```
Comment (brpowell) [inline .nvmrc:1]: Were there issues with the previous node version?

Comment (smithgp) [reply .nvmrc:1]: TBH, I didn't try, I just saw the mismatch from the volta config and switched it.
I just did `yarn install && yarn run build && yarn run test` with 10.15.3 and it was fine.
In any case, the versions in `.nvmrc` and `volta` in `package.json` (and whatever circleci does) should probably line up.

Comment (brpowell) [reply .nvmrc:1]: Ah yeah that makes sense. At first I thought this was a project template file heh.

**Review by brpowell — APPROVED** [review]

**Pull request merged**